src/crud/message.py

```
@@ -136,25 +138,63 @@ async def create_messages(
+
+            # Upsert vectors to external vector store with retry
+            if vector_records:
+                try:
+                    async for attempt in AsyncRetrying(
+                        stop=stop_after_attempt(3),
+                        wait=wait_exponential(multiplier=0.5, min=0.5, max=2.0),
+                        reraise=True,
+                    ):
+                        with attempt:
+                            await vector_store.upsert_many(namespace, vector_records)
+                except Exception as e:
```

Ruff (0.14.7)

Do not catch blind exception: `Exception`

(BLE001)